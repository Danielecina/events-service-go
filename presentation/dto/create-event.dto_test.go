package dto

import (
	"encoding/json"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestCreateEventRequest(t *testing.T) {
	t.Run("Expect to serialize CreateEventRequest correctly", func(t *testing.T) {
		input := CreateEventRequest{
			Name:        "Evento",
			Description: "Descrizione",
			Location:    "Luogo",
			UserID:      "user-1",
		}
		marshalInput, err := json.Marshal(input)
		require.NoError(t, err)
		expectedStr := string(marshalInput)
		snaps.MatchSnapshot(t, expectedStr, "create_event_request.json")
	})
}

func TestCreateEventErrorResponse(t *testing.T) {
	t.Run("Expect to serialize CreateEventErrorResponse correctly", func(t *testing.T) {
		errResp := CreateEventErrorResponse{
			Message:    "errore",
			ErrorCode:  "E001",
			StatusCode: 400,
			Errors:     []string{"campo mancante"},
		}
		marshalInput, err := json.Marshal(errResp)
		require.NoError(t, err)
		expectedStr := string(marshalInput)
		snaps.MatchSnapshot(t, expectedStr, "create_event_error_response.json")
	})
}
