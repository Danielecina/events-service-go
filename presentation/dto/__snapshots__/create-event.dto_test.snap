
[TestCreateEventRequest/Expect_to_serialize_CreateEventRequest_correctly - 1]
{"name":"Evento","description":"Descrizione","location":"Luogo","user_id":"user-1"}
create_event_request.json
---

[TestCreateEventErrorResponse/Expect_to_serialize_CreateEventErrorResponse_correctly - 1]
{"message":"errore","error_code":"E001","status_code":400,"errors":["campo mancante"]}
create_event_error_response.json
---
